test/remote/auth.ts

```diff
@@ -1,112 +1,125 @@
     },
     Storage: {
-      region: config.Region,
       bucket: config.Bucket,
       identityPoolId: config.IdentityPoolId,
-    },
-    API: {
-      endpoints: [
-        {
-          name: 'fotos',
-          endpoint: config.ServiceEndpoint,
-          region: config.Region,
-        },
-      ],
+      level: "protected",
+      region: config.Region,
     },
   });
 }
-function authenticateNewUser(config) {
+function authenticateNewUser(config: any) {
   return new Promise((resolve, reject) => {
-    const cognitoISP = new AWS.CognitoIdentityServiceProvider({
+    const cognitoISP = new CognitoIdentityServiceProvider({
       region: config.Region,
     });
 
-    const username = process.env.TEST_USER_NAME;
+    const username: string = process.env.TEST_USER_NAME || "";
 
-    const tempPassword = '@This1sChanged';
+    const tempPassword = "@This1sChanged";
 
     const params = {
-      UserPoolId: config.UserPoolId,
-      Username: username,
       DesiredDeliveryMediums: [
-        'EMAIL',
+        "EMAIL",
       ],
       ForceAliasCreation: true,
-      MessageAction: 'SUPPRESS',
+      MessageAction: "SUPPRESS",
       TemporaryPassword: tempPassword,
       UserAttributes: [
         {
-          Name: 'email',
+          Name: "email",
           Value: process.env.TEST_USER_EMAIL,
         },
       ],
+      UserPoolId: config.UserPoolId,
+      Username: username,
     };
     cognitoISP.adminCreateUser(params, (err, data) => {
       if (err) {
         reject(err);
       } else {
+        // tslint:disable-next-line:no-console
         console.log(JSON.stringify(data, null, 2));
         configureAmplify(config);
         Amplify.Auth.signIn(username, tempPassword)
-          .then(user => Amplify.Auth.completeNewPassword(user, process.env.TEST_USER_PWD))
+          .then((user: string) => Amplify.Auth.completeNewPassword(user, process.env.TEST_USER_PWD))
           .then(resolve)
           .catch(reject);
       }
     });
   });
 }
 
-function authenticateExistingUser(config) {
+function authenticateExistingUser(config: any) {
   return new Promise((resolve, reject) => {
     configureAmplify(config);
     Amplify.Auth.signIn(process.env.TEST_USER_NAME, process.env.TEST_USER_PWD)
       .then(resolve)
       .catch(reject);
   });
 }
-export function getIdentityId(response) {
+export function getIdentityId(response: any) {
   return Amplify.Auth.currentUserCredentials()
-    .then(res => ({
-      userIdentityId: res.params.IdentityId,
-      ...response,
-    }));
+    .then((res: any) => {
+      // forcibly assign the cognito credentials to aws-sdk
+      // storage doesnt pick up the creds from auth, probably because amplify isnt designed for node.js
+      // and competing aws-sdk dependencies may mean storage looks in the wrong
+      // aws-sdk lib version and reverts to tthe creds in the bash profile
+      // at least thats my best explanation so far.
+      AWS.config.credentials = res;
```

Comment:
Fix Amplify Storage auth issue by manually assigning Cognito creds to AWS.config